### PR TITLE
Raise proper error when deconding 16-bits jpegs

### PR DESCRIPTION
**Pull request by cyyever**
After some investigation, I found that #3613 failed because the image is 16 bit. However our decoding code assumes the image is 8-bit. So our allocated tensor size is too small to hold the decoded data and leads to memory overflow.

However, I can't find an easy way to decode 16-bit images without breaking the current decoding logic(we would have to change too many code places). So I simply issue an error.

Note: In order to support 16-bit png images, we should allocate the tensor with torch::kI16 and change 
```
auto datap = data.accessor<unsigned char, 1>().data();
```  
and 
```
auto ptr = tensor.accessor<uint8_t, 3>().data();
```
and the logic of read_callback. And although #4051 didn't fix this bug, I still think it's useful..

### CODE REVIEW

**Review by NicolasHug — APPROVED** [review]
Thanks @cyyever , LGTM!

I can confirm that the offending 16-bits jpeg now fails with the more informative error message rather than a segfault.

@fmassa should we open an issue to keep track of the 16-bits support, or is this not something we want?

**Comment by cyyever** [comment]
And there may be 32-bit or 64-bit png images. And I'm not sure the case of jpg.

File: torchvision/csrc/io/image/cpu/decode_png.cpp [hunk 1]
```diff
@@ -71,6 +71,11 @@ torch::Tensor decode_png(const torch::Tensor& data, ImageReadMode mode) {
     TORCH_CHECK(retval == 1, "Could read image metadata from content.")
   }
 
+  if (bit_depth != 8) {
```
Comment (NicolasHug) [inline torchvision/csrc/io/image/cpu/decode_png.cpp:74]: Should this be > 8?
looks like we have tests for 1-bit png that are failing now https://app.circleci.com/pipelines/github/pytorch/vision/9243/workflows/a1127581-3c33-48db-b2fb-f619e99e565d/jobs/677291/tests#failed-test-0

Comment (cyyever) [reply torchvision/csrc/io/image/cpu/decode_png.cpp:74]: Right, I am at supermarket, wait…

**Comment by NicolasHug** [comment]
Thanks @cyyever !

**Comment by fmassa** [comment]
Thanks for the fix @cyyever !

**Pull request merged**